pkg/prometheus/promcfg.go

```diff
@@ -1823,8 +1832,9 @@ func (cg *ConfigGenerator) generateAlertmanagerConfig(alerting *monitoringv1.Ale
 	}
 
 	alertmanagerConfigs := make([]yaml.MapSlice, 0, len(alerting.Alertmanagers))
-	s := store.ForNamespace(cg.prom.GetObjectMeta().GetNamespace())
 	for i, am := range alerting.Alertmanagers {
+		s := store.ForNamespace(*am.Namespace)
```

Comment:
Hmm, it's changing the current behavior which is to look for references into the Prometheus namespace.